tests/fft.hpp

```diff
@@ -20,198 +20,106 @@ using DDom = ddc::DiscreteDomain<DDim...>;
-                                   -(Kokkos::pow(coordinate(ddc::select<DFDim<ddc::Fourier<X>>>(e)), 2)
-                                     + ...)
-                                   / 2)
-                        << ", ";
-            });
-#endif
 
-    ddc::Chunk FFf_host_alloc
-            = ddc::Chunk(ddc::get_domain<DDim<X>...>(FFf), ddc::HostAllocator<Tin>());
+    ddc::Chunk FFf_host_alloc(FFf.domain(), ddc::HostAllocator<Tin>());
     ddc::ChunkSpan FFf_host = FFf_host_alloc.span_view();
     ddc::deepcopy(FFf_host, FFf);
-#if 0
-    std::cout << "\n iFFT(FFT):\n";
-    ddc::for_each(
-            ddc::policies::serial_host,
-            ddc::get_domain<DDim<X>...>(FFf_host),
-            [=](DElem<DDim<X>...> const e) {
-                (std::cout << ... << coordinate(ddc::select<DDim<X>>(e)))
-                        << "->" << Kokkos::abs(FFf_host(e)) << " " << Kokkos::abs(f_host(e))
-                        << ", ";
-            });
-#endif
 
     double criterion = Kokkos::sqrt(ddc::transform_reduce(
-            ddc::get_domain<DFDim<ddc::Fourier<X>>...>(Ff_host),
+            Ff_host.domain(),
             0.,
             ddc::reducer::sum<double>(),
             [=](DElem<DFDim<ddc::Fourier<X>>...> const e) {
-                return Kokkos::
-                               pow(Kokkos::abs(Ff_host(e))
-                                           - Kokkos::exp(
-                                                   -(Kokkos::
-                                                             pow(coordinate(ddc::select<
-                                                                            DFDim<ddc::Fourier<X>>>(
-                                                                         e)),
-                                                                 2)
-                                                     + ...)
-                                                   / 2),
-                                   2)
-                       / (LastSelector<std::size_t, X, X...>(Nx / 2, Nx) * ...);
+                double const xn2
+                        = (Kokkos::pow(ddc::coordinate(ddc::select<DFDim<ddc::Fourier<X>>>(e)), 2)
+                           + ...);
+                double const diff = Kokkos::abs(Ff_host(e)) - Kokkos::exp(-xn2 / 2);
+                std::size_t const denom
+                        = (ddc::detail::fft::LastSelector<std::size_t, X, X...>(Nx / 2, Nx) * ...);
+                return diff * diff / denom;
             }));
 
     double criterion2 = Kokkos::sqrt(ddc::transform_reduce(
-            ddc::get_domain<DDim<X>...>(FFf_host),
+            FFf_host.domain(),
             0.,
             ddc::reducer::sum<double>(),
             [=](DElem<DDim<X>...> const e) {
-                return Kokkos::pow(Kokkos::abs(FFf_host(e)) - Kokkos::abs(f_host(e)), 2)
-                       / Kokkos::pow(Nx, sizeof...(X));
+                double const diff = Kokkos::abs(FFf_host(e)) - Kokkos::abs(f_host(e));
```

Comment:
@blegouix Are we sure we want the difference of the absolute values and not the absolute value of the difference ?

Reply:
Let us see that later